src/App.tsx

```diff
@@ -15,47 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reverse = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.currentTarget.classList.toggle('is-light');
+    setIsReversed(!isReversed);
+  };
+
+  const handleReset = () => {
+    return (
+      setSortType(SortType.NONE),
+      setIsReversed(false)
+    );
+  };
+
+  const visibleGoods = [...goodsFromServer];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 1;
```

Comment:
lets by default not change our array
```suggestion
        return 0;
```